### PR TITLE
Use InodeView in DefaultPermissionChecker

### PR DESCRIPTION
**Pull request by madanadit**

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java [hunk 1]
```diff
@@ -74,7 +75,7 @@ public void checkParentPermission(Mode.Bits bits, LockedInodePath inodePath)
 
     // collects existing inodes info on the path. Note that, not all the components of the path have
     // corresponding inodes.
-    List<Inode> inodeList = inodePath.getInodeList();
+    List<InodeView> inodeList = (List<InodeView>) (List<?>) inodePath.getInodeList();
```
Comment (aaudiber) [inline core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: can we avoid these casts?

Comment (madanadit) [reply core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: don't think so. can we?

Comment (aaudiber) [reply core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: what if we give `inodeList` the type `? extends InodeView`?

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Pull request merged**